### PR TITLE
binary exponentiation added

### PR DESCRIPTION
**Pull request by tk315**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Article on binary exponentiation added

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#252

### CODE REVIEW

**Comment by tk315** [comment]
Please review sir @HarshCasper @iamrajiv @ankitaggarwal23 @atarax665 @gat786 @nidh-ish

**Review by iamrajiv — APPROVED** [review]

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@tk315 Resolve the merge conflict.

File: docs/mathematics/binary-exponentiation.mdx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+```
+Power is: 8
+```
+
+## Complexity Analysis
+
+- Time Complexity: `O(log b)`, where`b` is the exponent
+
+- Space Complexity: `O(1)`
```
Comment (vybhav72954) [inline docs/mathematics/binary-exponentiation.mdx:164]: Credits?

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

File: docs/mathematics/binary-exponentiation.mdx (hunk 1 above)
Comment (tk315) [reply docs/mathematics/binary-exponentiation.mdx:164]: Sir both c++ and python code is added by me

Comment (vybhav72954) [reply docs/mathematics/binary-exponentiation.mdx:164]: Are they present in NeoAlgo?

Comment (tk315) [reply docs/mathematics/binary-exponentiation.mdx:164]: May Be
But in the contribution guidelines it is written that we may use code form neoalgo or write it by ourselves

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**